apollo-router/src/plugins/telemetry/formatters/json.rs

```diff
@@ -3,18 +3,277 @@ use std::fmt;
+                        let key_name = k.as_str();
+                        !key_name.starts_with(APOLLO_PRIVATE_PREFIX) && !["code.filepath", "code.namespace", "code.lineno", "thread.id", "thread.name"].contains(&key_name)
+                    }) {
+                        serializer.serialize_entry(key.as_str(), &value.as_str())?;
+                    }
+                }
+                // Get custom dynamic attributes
+                let custom_attributes = ext.get::<LogAttributes>().map(|attrs| attrs.get_attributes());
+                if let Some(custom_attributes) = custom_attributes {
+                    for (key, value) in custom_attributes {
+                        serializer.serialize_entry(key.as_str(), value)?;
+                    }
+                }
+            }
+            // We have fields for this span which are valid JSON but not an object.
+            // This is probably a bug, so panic if we're in debug mode
+            Ok(_) if cfg!(debug_assertions) => panic!(
+                "span '{}' had malformed fields! this is a bug.\n  error: invalid JSON object\n  fields: {:?}",
+                self.0.metadata().name(),
+                data
+            ),
+            // If we *aren't* in debug mode, it's probably best not to
+            // crash the program, let's log the field found but also an
+            // message saying it's type  is invalid
+            Ok(value) => {
+                serializer.serialize_entry("field", &value)?;
+                serializer.serialize_entry("field_error", "field was no a valid object")?
+            }
+            // We have previously recorded fields for this span
+            // should be valid JSON. However, they appear to *not*
+            // be valid JSON. This is almost certainly a bug, so
+            // panic if we're in debug mode
+            Err(e) if cfg!(debug_assertions) => panic!(
+                "span '{}' had malformed fields! this is a bug.\n  error: {}\n  fields: {:?}",
+                self.0.metadata().name(),
+                e,
+                data
+            ),
+            // If we *aren't* in debug mode, it's probably best not
+            // crash the program, but let's at least make sure it's clear
+            // that the fields are not supposed to be missing.
+            Err(e) => serializer.serialize_entry("field_error", &format!("{}", e))?,
+        };
+        serializer.serialize_entry("name", self.0.metadata().name())?;
+        serializer.end()
+    }
+}
+
+impl<S, N> FormatEvent<S, N> for Json
+where
+    S: Subscriber + for<'lookup> LookupSpan<'lookup>,
+    N: for<'writer> FormatFields<'writer> + 'static,
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        mut writer: Writer<'_>,
+        event: &Event<'_>,
+    ) -> fmt::Result
+    where
+        S: Subscriber + for<'a> LookupSpan<'a>,
+    {
+        let timestamp = chrono::Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true);
```

Comment:
maybe let's use https://docs.rs/time/latest/time/serde/rfc3339/index.html instead of adding the entire chrono crate